dmriprep/utils/tests/test_vectors.py

```diff
@@ -0,0 +1,192 @@
+
+
+def test_vector_checkers():
+    """Test for vector corruption."""
+    from dipy.data import fetch_sherbrooke_3shell
+    from os.path import expanduser, join
+    from dipy.io import read_bvals_bvecs
+    import itertools
+
+    fetch_sherbrooke_3shell()
+    home = expanduser('~')
+    basedir = join(home, '.dipy', 'sherbrooke_3shell')
+    fbval = join(basedir, 'HARDI193.bval')
+    fbvec = join(basedir, 'HARDI193.bvec')
+
+    b0_thr = 50
+
+    # load bvecs/bvals.
+    bvals, bvecs = read_bvals_bvecs(fbval, fbvec)
+
+    # Perform various corruption checks using synthetic corrupted bval-bvec.
+    bval_short = bvals[:-1]
+    bval_long = np.hstack([bvals, bvals[-1]])
+    bval_no_b0 = bvals.copy()
+    bval_no_b0[0] = 51
+    bval_odd_b0 = bvals.copy()
+    bval_odd_b0[np.where(abs(bval_odd_b0) == 0)] = 0.00000001
+    bvec_short = bvecs[:-1]
+    bvec_long = np.vstack([bvecs, 0.5*bvecs[-1]])
+    bvec_no_b0 = bvecs.copy()
+    bvec_no_b0[np.where(np.all(abs(bvec_no_b0) == 0, axis=1) == True)] = [1, 1, 1]
+    bvec_odd_b0 = bvecs.copy()
+    bvec_odd_b0[np.where(np.all(abs(bvec_odd_b0) == 0, axis=1) == True)] = [10, 10, 10]
+
+    bval_lists = [bvals, bval_short, bval_long, bval_no_b0, bval_odd_b0]
+    bvec_lists = [bvecs, bvec_short, bvec_long, bvec_no_b0, bvec_odd_b0]
+    for bval, bvec in list(itertools.product(bval_lists, bvec_lists)):
+        [bvecs_checked, bvals_checked] = vectors.check_corruption(bvec, bval, b0_thr)
+        if (bval is bvals) and (bvec is bvecs):
+            assert bvecs_checked is not None
+            assert bvals_checked is not None
+        else:
+            with pytest.raises(Exception) as e_info:
+                assert bvecs_checked is None
+                assert bvals_checked is None
+
+
+def test_hemisphericity():
+    """Test vector hemisphere coverage and second-level model setting based on that coverage."""
+    from dipy.data import fetch_sherbrooke_3shell
+    from dipy.io import read_bvals_bvecs
+    from os.path import expanduser, join
+
+    fetch_sherbrooke_3shell()
+    home = expanduser('~')
+    basedir = join(home, '.dipy', 'sherbrooke_3shell')
+    fbval = join(basedir, 'HARDI193.bval')
+    fbvec = join(basedir, 'HARDI193.bvec')
+
+    b0_thr = 50
+
+    # load bvecs/bvals.
+    bvals, bvecs = read_bvals_bvecs(fbval, fbvec)
+    [gtab, _] = vectors.make_gradient_table(bvals, bvecs, b0_thr)
+
+    # Check hemisphere coverage.
+    [is_hemi, pole] = vectors.is_hemispherical(np.round(gtab.bvecs, 8)[~(np.round(gtab.bvecs, 8) == 0).all(1)])
+
+    # Use sampling scheme to set second-level model for eddy correction (this step would conceivably apply to any
+    # non-FSL form of eddy correction as well.
+    slm = vectors.set_slm(gtab, is_hemi)
+    assert is_hemi is False
+    assert isinstance(pole, (np.ndarray, np.generic)) is True
+    assert slm == 'none'
+
+
+def test_clean_vecs():
```

Comment:
this test is misplaced. should go under the interfaces directory.

Reply:
Yep!